rust/agama-lib/share/storage.schema.json

```diff
@@ -63,6 +68,89 @@
         }
       }
     },
+    "mdRaidElement": {
+      "anyOf": [
+        { "$ref": "#/$defs/formattedMdRaid" },
+        { "$ref": "#/$defs/partitionedMdRaid" }
+      ]
+    },
+    "formattedMdRaid": {
+      "description": "MD RAID without a partition table (directly formatted).",
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["level", "filesystem"],
+      "properties": {
+        "alias": { "$ref": "#/$defs/alias" },
+        "name": { "$ref": "#/$defs/mdRaidName" },
+        "level": { "$ref": "#/$defs/mdRaidLevel" },
+        "parity": { "$ref": "#/$defs/mdRaidParity" },
+        "chunkSize": { "$ref": "#/$defs/sizeValue" },
+        "devices": { "$ref": "#/$defs/mdRaidDevices" },
+        "encryption": { "$ref": "#/$defs/encryption" },
+        "filesystem": { "$ref": "#/$defs/filesystem" }
+      }
+    },
+    "partitionedMdRaid": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["level"],
+      "properties": {
+        "alias": { "$ref": "#/$defs/alias" },
+        "name": { "$ref": "#/$defs/mdRaidName" },
+        "level": { "$ref": "#/$defs/mdRaidLevel" },
+        "parity": { "$ref": "#/$defs/mdRaidParity" },
+        "chunkSize": { "$ref": "#/$defs/sizeValue" },
+        "devices": { "$ref": "#/$defs/mdRaidDevices" },
+        "ptableType": { "$ref": "#/$defs/ptableType" },
+        "partitions": {
+          "type": "array",
+          "items": { "$ref": "#/$defs/partitionElement" }
+        }
+      }
+    },
+    "mdRaidName": {
+      "description": "MD base name.",
+      "type": "string",
+      "examples": ["system"]
+    },
+    "mdRaidLevel": {
+      "title": "MD level",
+      "enum": [
+        "raid0",
+        "raid1",
+        "raid5",
+        "raid6",
+        "raid10"
+      ]
+    },
+    "mdRaidParity": {
+      "title": "MD parity",
+      "description": "Only applies to raid5, raid6 and raid10",
+      "enum": [
+        "left_asymmetric",
```

Comment:
At the man page of `mdadm` the names uses a dash instead of an underscore. Not sure what is best (or whether we should support both).

> The layout of the RAID5  parity  block  can  be  one  of  left-asymmetric,  left-symmetric, right-asymmetric, right-symmetric, la, ra, ls, rs.  The default is left-symmetric.
> 
> It  is also possible to cause RAID5 to use a RAID4-like layout by choosing parity-first, or parity-last.

Reply:
In the schema we are using snake_case for the enum values:

~~~
        "id": {
          "title": "Partition id",
          "enum": [
            "linux",
            "swap",
            "lvm",
            "raid",
            "esp",
            "prep",
            "bios_boot"
          ]
~~~

~~~
    "encryptionSwap": {
      "description": "Swap encryptions.",
      "enum": ["protected_swap", "secure_swap", "random_swap"]
    },
~~~

I think it makes sense to use the same format as the rest.